File: frontend/src/features/users/userService.ts
import apiClient from '@/api/apiClient';
import { User, CreateUserRequest, UpdateUserRequest } from '@/types';

export const userService = {
    getUsers: async (): Promise<User[]> => {
        const response = await apiClient.get<User[]>('/users');
        return response.data;
    },

    getUserById: async (id: number): Promise<User> => {
        const response = await apiClient.get<User>(`/users/${id}`);
        return response.data;
    },

    createUser: async (data: CreateUserRequest): Promise<User> => {
        const response = await apiClient.post<User>('/users', data);
        return response.data;
    },

    updateUser: async (id: number, data: UpdateUserRequest): Promise<User> => {
        const response = await apiClient.put<User>(`/users/${id}`, data);
        return response.data;
    },

    deleteUser: async (id: number): Promise<void> => {
        await apiClient.delete(`/users/${id}`);
    },
};

File: backend/src/main/java/com/fintech/finpro/service/UserService.java
package com.fintech.finpro.service;

import com.fintech.finpro.dto.CreateUserRequest;
import com.fintech.finpro.dto.UpdateUserRequest;
import com.fintech.finpro.dto.UserDTO;

import java.util.List;

public interface UserService {
    List<UserDTO> getAllUsers();

    UserDTO getUserById(Long id);

    UserDTO createUser(CreateUserRequest request);

    UserDTO updateUser(Long id, UpdateUserRequest request);

    void deleteUser(Long id);
}
